### ISSUE TITLE
remove no-crateBin hack 

### ISSUE THREAD

**Issue by andir** [kolloch/crate2nix#61]
Currently crate2nix uses a [hack](https://github.com/kolloch/crate2nix/blame/5b19c1b14e1b0e5522c3e44e300d0b332dc939e7/crate2nix/templates/build.nix.tera#L89) to avoid building binary targets for some crates. It constructs an `crateBin` attribute with the value of `[{name = ","; path = ",";}]`. `buildRustCrate` ignored that for a while since it was parsing comma separated pairs of names and paths and thus treated those as empty "lines".

I am currently working on a cleanup/rewrite of the `buildRustCrate` infrastructure and moved the logic that was previously in bash to the Nix expression. We can now (or once that PR is merged) properly set an empty list if we do not want any binaries to be produces by `buildRustCrate`.

There is now another hack to support users of previously generated `crate2nix` expressions: https://github.com/NixOS/nixpkgs/pull/75563/files#diff-8a4eb609f35e1485456f9c5c8ced5848R66-R69

I guess it is very unlikely that we can set a minimal version of nixpkgs for users of `crate2nix`? I also do not want to carry that hack in nixpkgs for several releases. What is your opinion on this?

PR: https://github.com/NixOS/nixpkgs/pull/75563

**Comment by kolloch** [comment]
Thank you for reaching out! What about ignoring entries with a name with a comma? That should be safe.

**Comment by andir** [comment]
That is what is currently implemented in Nixpkgs. I just ignores those crates. It would be nice to have some deprecation in the future where we can get rid of this hack completely.

**Comment by kolloch** [comment]
I documented our dependency on a recent nixpkgs version. In practice, this has been my policy anyhow.

`buildRustCrate` is still too immature to depend only on the stable version. In fact, I even considered depending on a forked version but I like that `buildRustCrate` is shared between `carnix` and `crate2nix` and therefore maybe gets more attention.
